TaskModules/powershell/Azure/SPNCreation.ps1

```diff
@@ -42,6 +135,32 @@ function Get-Password
     }
 }
 
+
+if ($environmentName -like $AZURESTACK_ENVIRONMENT)
+{
+    if(-not $azureStackManagementURL)
+    {
+        $azureStackEnvironment = Get-AzureRmEnvironment -Name $AZURESTACK_ENVIRONMENT
+        if(-not $azureStackEnvironment)
+        {
+            throw "AzureStack Enviornment is not present in AzureRmEnvironments in current PS Session. Please provide AzureStackManagementURL as argument or add AzureStack to AzureRmEnvironments manually."
+        }
+    }
+    else
+    {
+        try
+        {
+            Add-AzureStackToAzureRmEnvironment -EndpointURI $azureStackManagementURL -Name $AZURESTACK_ENVIRONMENT
+        }
+        catch
+        {
+            Write-Output "ERROR: Failed to add AzureStack environment to AzureRmEnvironments";
+            throw $_.Exception
+        }
+    }
+}
+
+
 #Initialize
 $ErrorActionPreference = "Stop"

```

Comment:
Your Error action preference and Verbose preference should be set before you initialize AzureStack environment